### ISSUE TITLE
OverlayTutorialScope causing render error

### ISSUE THREAD

**Issue by hashirshoaeb** [TabooSun/overlay_tutorial#25]
#### Description
I have a list of cards. And I trigger tutorial when user swipes first time. But if user frequently swipes more than one card then tutorial causes error(see attached screenshots). 
#### Screenshots

![Screenshot 2021-07-01 at 5 07 13 PM](https://user-images.githubusercontent.com/35165481/124121816-c9c0d080-da8e-11eb-8e5b-6ab21a19dd9d.png)
![Screenshot 2021-07-01 at 5 08 36 PM](https://user-images.githubusercontent.com/35165481/124121993-fb399c00-da8e-11eb-8986-af19d13b17a5.png)

#### Platform: 
- [x] iOS
- [x] Android

#### Flutter doctor -v 
```bash
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale en-GB)
    • Flutter version 2.0.5 at /Users/hashirshoaeb/flutter2.0
    • Framework revision adc687823a (3 months ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/hashirshoaeb/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1
```

**Comment by TabooSun** [comment]
It seems that you are using the old version, can you update the package to 2.0.4 and see if the issue persists?

**Comment by Siqlain-Hanif** [comment]
@TabooSun seems like that particular problem is fixed in 2.0.4, previously we were using 2.0.1. 
Thanks for the response.
